dashboard/app/models/pd/application/teacher2021_application.rb

```diff
@@ -0,0 +1,779 @@
+      update_form_data_hash(
+        {
+          principal_response_first_name: principal_response[:first_name],
+          principal_response_last_name: principal_response[:last_name],
+          principal_response_email: principal_response[:email],
+          principal_school_name: principal_school.try(:name) || principal_response[:school_name],
+          principal_school_type: principal_school.try(:school_type),
+          principal_school_district: principal_school.try(:district).try(:name),
+          principal_approval: principal_response.values_at(:do_you_approve, :do_you_approve_other).compact.join(" "),
+          principal_plan_to_teach: principal_response.values_at(:plan_to_teach, :plan_to_teach_other).compact.join(" "),
+          principal_schedule_confirmed: principal_response.values_at(:committed_to_master_schedule, :committed_to_master_schedule_other).compact.join(" "),
+          principal_implementation: implementation_string,
+          principal_total_enrollment: principal_response[:total_student_enrollment],
+          principal_diversity_recruitment: principal_response.values_at(:committed_to_diversity, :committed_to_diversity_other).compact.join(" "),
+          principal_free_lunch_percent: format("%0.02f%%", principal_response[:free_lunch_percent]),
+          principal_underrepresented_minority_percent: format("%0.02f%%", principal_approval.underrepresented_minority_percent),
+          principal_american_indian_or_native_alaskan_percent: format("%0.02f%%", principal_response[:american_indian]),
+          principal_asian_percent: format("%0.02f%%", principal_response[:asian]),
+          principal_black_or_african_american_percent: format("%0.02f%%", principal_response[:black]),
+          principal_hispanic_or_latino_percent: format("%0.02f%%", principal_response[:hispanic]),
+          principal_native_hawaiian_or_pacific_islander_percent: format("%0.02f%%", principal_response[:pacific_islander]),
+          principal_white_percent: format("%0.02f%%", principal_response[:white]),
+          principal_other_percent: format("%0.02f%%", principal_response[:other]),
+          principal_wont_replace_existing_course: replace_course_string,
+          principal_send_ap_scores: principal_response[:send_ap_scores],
+          principal_pay_fee: principal_response[:pay_fee],
+          principal_contact_invoicing: principal_response[:contact_invoicing],
+          principal_contact_invoicing_detail: principal_response[:contact_invoicing_detail]
+        }
+      )
+      save!
+      auto_score!
+      queue_email(:principal_approval_completed, deliver_now: true)
+      queue_email(:principal_approval_completed_partner, deliver_now: true)
+    end
+
+    # @override
+    def default_response_score_hash
+      {
+        meets_minimum_criteria_scores: {},
+        meets_scholarship_criteria_scores: {},
+        bonus_points_scores: {}
+      }
+    end
+  end
+end
```

Comment:
This is a huge change to make every year, even if it's essentially going to become a rename if/when the older form is deprecated.  Is there any way we can make this particular model more generic?